File: newsflow-android/app/src/main/java/com/newsflow/android/ui/screens/GeoData.kt
package com.newsflow.android.ui.screens

/**
 * Country + US-state option lists for the local-area form. Mirrors the web
 * LocationQuery constants (the server is the source of truth for validation).
 */
object GeoData {
    // US pinned first; the rest alphabetical by name.
    val COUNTRIES: List<Pair<String, String>> = listOf(
        "US" to "United States",
        "AU" to "Australia",
        "BR" to "Brazil",
        "CA" to "Canada",
        "FR" to "France",
        "DE" to "Germany",
        "IN" to "India",
        "IE" to "Ireland",
        "IT" to "Italy",
        "JP" to "Japan",
        "MX" to "Mexico",
        "NL" to "Netherlands",
        "NZ" to "New Zealand",
        "NG" to "Nigeria",
        "PH" to "Philippines",
        "SG" to "Singapore",
        "ZA" to "South Africa",
        "ES" to "Spain",
        "GB" to "United Kingdom",
        "KE" to "Kenya",
    )

    val US_STATES: List<Pair<String, String>> = listOf(
        "AL" to "Alabama", "AK" to "Alaska", "AZ" to "Arizona", "AR" to "Arkansas",
        "CA" to "California", "CO" to "Colorado", "CT" to "Connecticut", "DE" to "Delaware",
        "DC" to "District of Columbia", "FL" to "Florida", "GA" to "Georgia", "HI" to "Hawaii",
        "ID" to "Idaho", "IL" to "Illinois", "IN" to "Indiana", "IA" to "Iowa",
        "KS" to "Kansas", "KY" to "Kentucky", "LA" to "Louisiana", "ME" to "Maine",
        "MD" to "Maryland", "MA" to "Massachusetts", "MI" to "Michigan", "MN" to "Minnesota",
        "MS" to "Mississippi", "MO" to "Missouri", "MT" to "Montana", "NE" to "Nebraska",
        "NV" to "Nevada", "NH" to "New Hampshire", "NJ" to "New Jersey", "NM" to "New Mexico",
        "NY" to "New York", "NC" to "North Carolina", "ND" to "North Dakota", "OH" to "Ohio",
        "OK" to "Oklahoma", "OR" to "Oregon", "PA" to "Pennsylvania", "RI" to "Rhode Island",
        "SC" to "South Carolina", "SD" to "South Dakota", "TN" to "Tennessee", "TX" to "Texas",
        "UT" to "Utah", "VT" to "Vermont", "VA" to "Virginia", "WA" to "Washington",
        "WV" to "West Virginia", "WI" to "Wisconsin", "WY" to "Wyoming",
    )
}

File: newsroom-flow-ios/NewsroomFlow/Data/GeoData.swift
import Foundation

/// Country + US-state option lists for the local-area form. Mirrors the web
/// LocationQuery constants (the server validates authoritatively).
enum GeoData {
    // US pinned first; the rest alphabetical by name.
    static let countries: [(code: String, name: String)] = [
        ("US", "United States"),
        ("AU", "Australia"),
        ("BR", "Brazil"),
        ("CA", "Canada"),
        ("FR", "France"),
        ("DE", "Germany"),
        ("IN", "India"),
        ("IE", "Ireland"),
        ("IT", "Italy"),
        ("JP", "Japan"),
        ("KE", "Kenya"),
        ("MX", "Mexico"),
        ("NL", "Netherlands"),
        ("NZ", "New Zealand"),
        ("NG", "Nigeria"),
        ("PH", "Philippines"),
        ("SG", "Singapore"),
        ("ZA", "South Africa"),
        ("ES", "Spain"),
        ("GB", "United Kingdom"),
    ]

    static let usStates: [(code: String, name: String)] = [
        ("AL", "Alabama"), ("AK", "Alaska"), ("AZ", "Arizona"), ("AR", "Arkansas"),
        ("CA", "California"), ("CO", "Colorado"), ("CT", "Connecticut"), ("DE", "Delaware"),
        ("DC", "District of Columbia"), ("FL", "Florida"), ("GA", "Georgia"), ("HI", "Hawaii"),
        ("ID", "Idaho"), ("IL", "Illinois"), ("IN", "Indiana"), ("IA", "Iowa"),
        ("KS", "Kansas"), ("KY", "Kentucky"), ("LA", "Louisiana"), ("ME", "Maine"),
        ("MD", "Maryland"), ("MA", "Massachusetts"), ("MI", "Michigan"), ("MN", "Minnesota"),
        ("MS", "Mississippi"), ("MO", "Missouri"), ("MT", "Montana"), ("NE", "Nebraska"),
        ("NV", "Nevada"), ("NH", "New Hampshire"), ("NJ", "New Jersey"), ("NM", "New Mexico"),
        ("NY", "New York"), ("NC", "North Carolina"), ("ND", "North Dakota"), ("OH", "Ohio"),
        ("OK", "Oklahoma"), ("OR", "Oregon"), ("PA", "Pennsylvania"), ("RI", "Rhode Island"),
        ("SC", "South Carolina"), ("SD", "South Dakota"), ("TN", "Tennessee"), ("TX", "Texas"),
        ("UT", "Utah"), ("VT", "Vermont"), ("VA", "Virginia"), ("WA", "Washington"),
        ("WV", "West Virginia"), ("WI", "Wisconsin"), ("WY", "Wyoming"),
    ]
}
